docs/DeveloperGuide.md

```diff
@@ -1,38 +1,360 @@
+3) The `addModule` method returns the module that is added to the `Parser` class.
+4) The `getModuleListSize` method is called from `ModuleList` class.
+5) The `printAddedModule` method is called from the `Print` class to display the result to the user.
+
+The sequence of events above can be represented with the following sequence diagram:
+
+![addInputCommand](diagrams/AddInputCommand.png)
+
+<small><i>Figure ???</i></small>
+
+### 3.4. Find Modules
+
+The find modules features is facilitated by the `Parser`, `ModuleList` and `Module` Classes. `Parser` helps to check
+for user input to determine is user doing a search via a module name or module type, which will then invoke the 
+`findModuleByName` and `findModuleByType` methods respectively. If any modules are found, the ChatBot will print out
+all the modules, else the ChatBot will return a message to the user, saying that no modules were found with the search
+term they used.
+
+Given below is an example usage scenario and how the findModuleByName mechanism behaves at each step.
+
+Step 1: The user launches the application for the first time. The ChatBot will be initialised with an empty ArrayList
+of Modules.
+
+Step 2: The user executes `add /CS2113T CS2101 /4 /CORE /2 /2` to add two modules into the ChatBot via the addModules 
+feature
+
+The following object diagram shows the current state of the ChatBot:
+![FindModule](diagrams/FindTwoModules.png)
+
+<small><i>Figure ???</i></small>
+
+Step 3: The user now wants to find the details of the modules he has just added. He executes `find /name /CS` which
+will call `findModuleByName`. This method will then print out all the modules that have the keyword "CS" in their
+module name.
+
+The following sequence diagram shows how the `findModuleByName` operation works:
+
+![FindModule](diagrams/FindModule.png)
+
+<small><i>Figure ???</i></small>
+
+### 3.5. Delete Modules
+
+`delete` command:
+
+The `delete` command is used by the user to delete a module which is specified by the user.
+
+The sequence in which the `Parser` class handles the `delete` command is as follows:
+1) The `Parser` class extracts the necessary fields from the user input, and calls upon the `deleteModule`
+   method in the `ModuleList` class.
+2) The method then loops through the moduleList array, attempting to find a module with the module code
+   that was specified by the user.
+3) If a corresponding module is found, the `deleteModule` method will remove the module from the moduleList array
+   and return the deleted `Module` object. Otherwise, the `deleteModule` method returns `null`.
+4) If a deleted `Module` object is returned from `deleteModule`, the `Parser` class calls upon
+   `printDeletedModule()` of the `Print` class. Otherwise, the `Parser` class calls upon `printNoDeletedModuleFound()`
+   of the `Print` class. This displays to the user the result of the `delete` command.
+
+![deleteModule](diagrams/DeleteModule.png)
```

Comment:
Perhaps could shift the "[module not found]" in the alt frame more to the right as its overlapping with the activation bar?